tests/test_api/test_methods/test_get_url.py

```diff
@@ -46,5 +46,61 @@ def test_method(
 
         if expected_result is None:
             assert fake_message.get_url(force_private=force_private) is None
+            assert (
+                fake_message.get_url(force_private=force_private, include_thread_id=True) is None
+            )
         else:
             assert fake_message.get_url(force_private=force_private) == expected_result
+            assert (
+                fake_message.get_url(force_private=force_private, include_thread_id=True)
+                == expected_result
+            )
+
+    @pytest.mark.parametrize(
+        "chat_type,chat_id,chat_username,force_private,include_thread_id,fake_thread_id_topic,expected_result",
+        [
+            ["supergroup", -100123456789, None, False, False, None, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, False, False, 3, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, False, True, None, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, False, True, 3, "https://t.me/c/123456789/3/10"],
+            ["supergroup", -100123456789, None, True, False, None, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, True, False, 3, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, True, True, None, "https://t.me/c/123456789/10"],
+            ["supergroup", -100123456789, None, True, True, 3, "https://t.me/c/123456789/3/10"],
+            ["supergroup", -100123456789, "name", False, False, None, "https://t.me/name/10"],
+            ["supergroup", -100123456789, "name", False, False, 3, "https://t.me/name/10"],
+            ["supergroup", -100123456789, "name", False, True, None, "https://t.me/name/10"],
+            ["supergroup", -100123456789, "name", False, True, 3, "https://t.me/name/3/10"],
+            ["supergroup", -10012345678, "name", True, False, None, "https://t.me/c/12345678/10"],
+            ["supergroup", -100123456789, "name", True, False, 3, "https://t.me/c/123456789/10"],
+            ["supergroup", -10012345678, "name", True, True, None, "https://t.me/c/12345678/10"],
+            ["supergroup", -100123456789, "name", True, True, 3, "https://t.me/c/123456789/3/10"],
```

Comment:
For what purpose `chat_type` and `chat_id` are parametrized here?

Reply:
This creates different types of updates.
I planned that a “channel” type could be added here in the future. Although, as I understand it, the channels do not have topics.